osm2streets/src/types.rs

```diff
@@ -120,52 +120,36 @@ pub enum ConflictType {
     Cross,
 }
 
-/// What kind of feature an Intersection actually represents. Any connection between roads in the
-/// network graph is represented by an Intersection, but many of them are not "intersections" in
-/// the traditional sense.
-///
-/// This type might be better named IntersectionType and the distinction between MultiConnection,
-/// Merge and Crossing is of dubious value and will probably change in the future.
+/// What kind of feature an `Intersection` actually represents. Any connection between roads in the
+/// network graph is represented by an `Intersection`, but many of them are not traffic
+/// "intersections" in the common sense.
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
-pub enum IntersectionComplexity {
-    /// The edge of the data that we have.
-    ///
-    /// The road may continue past here in reality and do anything, but the current dataset is clipped here.
+pub enum IntersectionType {
+    /// A `Road` ends because the road crosses the map boundary.
     MapEdge,
 
-    /// Two roads connect, one ending where the other begins, without conflict
-    ///
-    /// Like a road way has been sliced in two, because a lane was added.
-    Connection,
-
-    /// Multiple road ways connect, where the carriageway splits, where a road way joins with or
-    /// diverges from another without conflict.
-    ///
-    /// For example, when a single carriageway splits into a dual carriageway,
-    /// or when highway entrances and exits join and leave a highway.
+    /// A single `Road` ends because the actual roadway ends; "the end of the line".
     ///
-    /// You would expect no lane marking to be interupted, and maybe even a buffer area painted
-    /// where the lanes meet each other.
-    MultiConnection,
+    /// E.g. turning circles, road end signs, train terminus thingos, ...
+    Terminus,
 
-    /// One or more "minor" roads merge into (yielding) or out of a "major" road,
-    /// which retains its right of way.
+    /// Multiple `Road`s connect but no flow of traffic interacts with any other.
     ///
-    /// You would expect lane markings for the major road to continue (largely)
-    /// uninterrupted through the intersection, with a yield line ending the minor road.
-    Merge,
+    /// Usually one `Road` ends and another begins because the number of lanes has changed or some
+    /// other attribute of the roadway has changed. More than two `Road`s could be involved,
+    /// e.g. when a single carriageway (a bidirectional `Road`) splits into a dual carriageway
+    /// (two oneway `Road`s).
```

Comment:
The "no flow of traffic interacts" part means that if we had a DC rejoining in a place where U-turns were explicitly allowed, it'd be an `Intersection` then?

Reply:
Yes, that's the spirit of it. I would expect some additional markings or something in that situation. In Perth I would never expect that, there would be a dedicated [U-turn lane](https://www.openstreetmap.org/way/669734507) or [dedicated area in the median](https://www.openstreetmap.org/way/29018596) for those manoeuvres.

If it turns out U-turns are allowed in situations that we still want to treat like `Connections`, then maybe U-turns become a special case.